src/FontManager.ts

```diff
@@ -74,35 +76,36 @@ export class FontManager extends EventDispatcher<EventMap> {
 
   update(newChars: string): void {
     let needsUpdate = false;
+    let newAlphabet = "";
     for (const char of newChars) {
       if (!this.alphabet.includes(char)) {
-        this.alphabet += char;
+        newAlphabet += char;
         needsUpdate = true;
       }
     }
 
     if (!needsUpdate) {
       return;
     }
-    const atlasWidth = 1024;
-    const atlasHeight = 1024;
-    const atlas = new Uint8ClampedArray(atlasWidth * atlasHeight);
+    const atlasWidth = this.atlasData.width;
+    const atlasHeight = this.atlasData.height;
+    const atlas = this.atlasData.data;
     const fontSize = this.options.fontSize ?? 48;
     const buffer = Math.ceil(fontSize / 16);
-    const tinysdf = new TinySDF({
-      fontSize,
-      buffer,
-      radius: Math.ceil(fontSize / 4),
-      fontFamily: this.options.fontFamily ?? "monospace",
-    });
+    const fontFamily = this.options.fontFamily ?? "monospace";
+    let tinysdf = this.#tinysdfCache[`${fontSize}/${fontFamily}`];
+    if (tinysdf == undefined) {
+      tinysdf = new TinySDF({
+        fontSize,
+        buffer,
+        radius: Math.ceil(fontSize / 4),
+        fontFamily,
+      });
+      this.#tinysdfCache[`${fontSize}/${fontFamily}`] = tinysdf;
```

Comment:
Not sure how I feel about holding onto these indefinitely, what do you think of just tracking the last used size & family and only keeping the most recent tinysdf instance?

....and actually now that I look at it, I'm not sure these ever change dynamically. The options are passed in the constructor but it looks like we don't change them. Maybe we should make the options readonly and then we don't need to worry about this.